internal/monitors/resourcesync/resourcesync.go

```diff
@@ -0,0 +1,289 @@
+		for _, genResource := range link.resources {
+			err = r.applyResource(genResource, &link.resourceSync)
+			if err != nil {
+				log.Errorf("resourcesync/%s: Failed to apply resource. Error: %s", link.resourceSync.Name, err.Error())
+				break
+			}
+		}
+		if err != nil {
+			break
+		}
+		link.resourceSync.Status.Data.LastSyncedCommitHash = &link.hash
+		err = r.store.GetResourceSyncStore().UpdateStatusIgnoreOrg(&link.resourceSync)
+		if err != nil {
+			log.Errorf("Failed to update resourcesync status for %s: %v", link.resourceSync.Name, err)
+		}
+	}
+}
+func (r *ResourceSync) cloneRepo(repo *model.Repository, revision *string) (billy.Filesystem, string, error) {
+	storage := gitmemory.NewStorage()
+	mfs := memfs.New()
+	ops := &git.CloneOptions{
+		URL:   *repo.Spec.Data.Repo,
+		Depth: 1,
+	}
+	if repo.Spec.Data.Username != nil && repo.Spec.Data.Password != nil {
+		ops.Auth = &githttp.BasicAuth{
+			Username: *repo.Spec.Data.Username,
+			Password: *repo.Spec.Data.Password,
+		}
+	}
+	if revision != nil {
+		ops.ReferenceName = gitplumbing.ReferenceName(*revision)
+	}
+	gitRepo, err := git.Clone(storage, mfs, ops)
+	if err != nil {
+		return nil, "", err
+	}
+	head, err := gitRepo.Head()
+	if err != nil {
+		return nil, "", err
+	}
+	hash := head.Hash().String()
+	return mfs, hash, nil
+}
+
+func (r *ResourceSync) extractResourcesFromDir(orgId string, mfs billy.Filesystem, path string) ([]genericResource, error) {
+	genericResources := []genericResource{}
+	files, err := mfs.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+	for _, file := range files {
+		if !file.IsDir() && isValidFile(file.Name()) { // Not going recursivly into subfolders
+			resources, err := r.extractResourcesFromFile(orgId, mfs, mfs.Join(path, file.Name()))
+			if err != nil {
+				return nil, err
+			}
+			genericResources = append(genericResources, resources...)
+		}
+	}
+	return genericResources, nil
+}
+
+func (r *ResourceSync) extractResourcesFromFile(orgId string, mfs billy.Filesystem, path string) ([]genericResource, error) {
+	genericResources := []genericResource{}
+
+	file, err := mfs.Open(path)
+	if err != nil {
+		// Failed to open file....
+		return nil, err
+	}
+	defer file.Close()
+	decoder := yamlutil.NewYAMLOrJSONDecoder(file, 100)
+
+	for {
+		var resource genericResource
+		err = decoder.Decode(&resource)
+		if err != nil {
+			break
+		}
+		genericResources = append(genericResources, resource)
+	}
+	if !errors.Is(err, io.EOF) {
+		return nil, err
+	}
+	return genericResources, nil
+
+}
+
+func (r *ResourceSync) applyResource(resource genericResource, rs *model.ResourceSync) error {
+	kind, ok := resource["kind"].(string)
+	rsName := fmt.Sprintf("resourcesync/%s", rs.Name)
+	orgID := rs.OrgID
+	if !ok {
+		return fmt.Errorf("%s: skipping resource of unspecified kind: %v", rsName, resource)
+	}
+	buf, err := json.Marshal(resource)
+	if err != nil {
+		return fmt.Errorf("%s: skipping resource of kind %q: %v", rsName, kind, err)
```

Comment:
good practice to wrap all errors this can be useful in testing etc.
```suggestion
		return fmt.Errorf("%s: skipping resource of kind %q: %w", rsName, kind, err)
```